plugins/net_plugin/net_plugin.cpp

```diff
@@ -4040,92 +3948,265 @@ namespace eosio {
       FC_CAPTURE_AND_RETHROW()
    }
 
-   /**
-    *  Used to trigger a new connection from RPC API
-    */
+   /// RPC API
    string net_plugin::connect( const string& host ) {
-      return my->connect( host );
+      return my->connections.connect( host );
    }
 
-   string net_plugin_impl::connect( const string& host ) {
-      std::lock_guard<std::shared_mutex> g( connections_mtx );
-      if( find_connection( host ) )
-         return "already connected";
+   /// RPC API
+   string net_plugin::disconnect( const string& host ) {
+      return my->connections.disconnect(host);
+   }
 
-      connection_ptr c = std::make_shared<connection>( host );
-      fc_dlog( logger, "calling active connector: ${h}", ("h", host) );
-      if( c->resolve_and_connect() ) {
-         fc_dlog( logger, "adding new connection to the list: ${host} ${cid}", ("host", host)("cid", c->connection_id) );
-         c->set_heartbeat_timeout( heartbeat_timeout );
-         connections.insert( c );
+   /// RPC API
+   std::optional<connection_status> net_plugin::status( const string& host )const {
+      return my->connections.status(host);
+   }
+
+   /// RPC API
+   vector<connection_status> net_plugin::connections()const {
```

Comment:
Can the method name be changed to status_list or some plural?

Reply:
No, because it matches the API name.